### PR TITLE
Support `=~` operator 

### LINKED ISSUE #28: [node-v6] Got SyntaxError parsing to JS on Node v6
I've discovered one more failing use case:

CartoCSS:

```
#layer0['mapnik::geometry_type'=1] { 
  marker-fill: red;
  marker-width: 10;
}
```

```
> if((8388607 & (1 << ctx.zoom)) && data.mapnik::geometry_type === 1){_value = '#ff0000';}
                                                
SyntaxError: Unexpected token :
    at Object._createFn (/vagrant/repos/Windshaft/node_modules/carto/lib/carto/renderer_js.js:153:12)
    at Object.parse (/vagrant/repos/Windshaft/node_modules/carto/lib/carto/renderer_js.js:245:40)
    at Object.setStyle (/vagrant/repos/Windshaft/node_modules/carto/lib/carto/renderer_js.js:120:23)
    at Object.CartoCSS (/vagrant/repos/Windshaft/node_modules/carto/lib/carto/renderer_js.js:10:10)
    at render (/vagrant/repos/Windshaft/node_modules/carto/lib/carto/renderer_js.js:280:12)
    at Object.module.exports.optionsFromCartoCSS (/vagrant/repos/Windshaft/lib/windshaft/utils/cartocss_utils.js:13:41)
    at /vagrant/repos/Windshaft/lib/windshaft/renderers/mapnik/geojson_renderer.js:20:38
...
```

Also it just happens on Node v6.

### PR DESCRIPTION
**Pull request by dgaubert**
Fix #28.

### CODE REVIEW

File: test/rendering_js.test.js [hunk 1]
```diff
@@ -4,24 +4,24 @@ var carto = require('../lib/carto');
 describe('RenderingJS', function() {
   var shader;
   var style = [
-  '#world {',
```
Comment (rochoa) [inline test/rendering_js.test.js]: Can we keep this unmodified?

File: test/rendering_js.test.js [hunk 2]
```diff
@@ -4,24 +4,24 @@ var carto = require('../lib/carto');
...
+  '}',
+  '',
+  '#worls[frame-offset = 10] {',
+      'line-width: 4;',
   '}'
   ].join('\n');
 
   beforeEach(function() {
-    shader = (new carto.RendererJS({ debug: true })).render(style);
+    shader = (new carto.RendererJS({ debug: false })).render(style);
```
Comment (rochoa) [inline test/rendering_js.test.js]: I would keep it as it was, or at least do it in a separate commit.

File: lib/carto/tree/filterset.js [hunk 3]
```diff
@@ -92,6 +92,9 @@ tree.Filterset.prototype.toJS = function(env) {
       val = filter._val.toString(true);
     }
     var attrs = "data";
+    if (op === '=~') {
+      return "(typeof " + attrs + "['" + filter.key.value  + "'] === 'string') && " + attrs + "['" + filter.key.value  + "'].match(" + (val.is === 'string' ? "'" + val.toString().replace(/'/g, "\\'") + "'" : val) + ")";
```
Comment (rochoa) [inline lib/carto/tree/filterset.js]: Do we need to check the type here? I guess that should be validated in other place.

Comment (rochoa) [inline lib/carto/tree/filterset.js]: Either that or cast `data[filter.key.value]` as a string, as the problem is to use `.match` on a Number. If you test it, you can see that is possible to use `=~` on numbers.

Comment (rochoa) [inline lib/carto/tree/filterset.js]: screen shot 2017-03-03 at 16 21 38

File: test/rendering_js.test.js (hunk 1 above)
Comment (dgaubert) [inline test/rendering_js.test.js]: 👍

File: test/rendering_js.test.js (hunk 2 above)
Comment (dgaubert) [inline test/rendering_js.test.js]: 👍

File: lib/carto/tree/filterset.js (hunk 3 above)
Comment (dgaubert) [inline lib/carto/tree/filterset.js]: :man_facepalming:

File: lib/carto/tree/filterset.js [hunk 4]
```diff
@@ -93,7 +93,7 @@ tree.Filterset.prototype.toJS = function(env) {
     }
     var attrs = "data";
     if (op === '=~') {
-      return "(typeof " + attrs + "['" + filter.key.value  + "'] === 'string') && " + attrs + "['" + filter.key.value  + "'].match(" + (val.is === 'string' ? "'" + val.toString().replace(/'/g, "\\'") + "'" : val) + ")";
+      return "(" + attrs + "['" + filter.key.value  + "'] + '').match(" + (val.is === 'string' ? "'" + val.toString().replace(/'/g, "\\'") + "'" : val) + ")";
```
Comment (rochoa) [inline lib/carto/tree/filterset.js]: 👍

**Comment by rochoa** [comment]
🇪🇸 🇨🇭 :shipit: .

**Comment by dgaubert** [comment]
flag_of_la_rioja_ with_coat_of_arms svg
